plugins/account_history/plugin.cpp

```diff
@@ -347,50 +353,38 @@ std::string to_string(operation_direction_type dir) {
 
         void operator()(const change_recovery_account_operation& op) {
             impacted.insert(make_pair(op.account_to_recover, operation_direction_type::any));
+            impacted.insert(make_pair(op.new_recovery_account, operation_direction_type::any));
         }
 
-        void operator()(const escrow_transfer_operation& op) {
+        template<class op_type>
+        void insert_from_to_agent_direction(const op_type& op) {
             impacted.insert(make_pair(op.from, operation_direction_type::sender));
             impacted.insert(make_pair(op.to, operation_direction_type::receiver));
             impacted.insert(make_pair(op.agent, operation_direction_type::receiver));
         }
 
+        void operator()(const escrow_transfer_operation& op) {
+            insert_from_to_agent_direction(op);
+        }
+
         void operator()(const escrow_approve_operation& op) {
-            impacted.insert(make_pair(op.from, operation_direction_type::sender));
-            impacted.insert(make_pair(op.to, operation_direction_type::receiver));
-            impacted.insert(make_pair(op.agent, operation_direction_type::receiver));
+            insert_from_to_agent_direction(op);
         }
 
         void operator()(const escrow_dispute_operation& op) {
-            impacted.insert(make_pair(op.from, operation_direction_type::sender));
-            impacted.insert(make_pair(op.to, operation_direction_type::receiver));
-            impacted.insert(make_pair(op.agent, operation_direction_type::receiver));
+            insert_from_to_agent_direction(op);
         }
 
         void operator()(const escrow_release_operation& op) {
-            impacted.insert(make_pair(op.from, operation_direction_type::sender));
-            impacted.insert(make_pair(op.to, operation_direction_type::receiver));
-            impacted.insert(make_pair(op.agent, operation_direction_type::receiver));
+            insert_from_to_agent_direction(op);
```

Comment:
note: the initiator (signer) of `escrow_dispute_operation ` is `who`. he can be either `from` or `to` or `agent`.
the `receiver` receives funds, he can be either `from` or `to`